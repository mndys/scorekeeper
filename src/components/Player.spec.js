import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Player from './Player'

describe('Player', () => {
  it('has a name, two buttons and a score', () => {
    render(<Player name="John" score={22} />)
    expect(screen.getByText('John')).toBeInTheDocument()
    expect(screen.getAllByRole('button', /[+-]/)).toHaveLength(2)
    expect(screen.getByText('22')).toBeInTheDocument()
  })

  it('supports callbacks for + and - buttons', () => {
    const onPlusCallback = jest.fn()
    const onMinusCallback = jest.fn()
    render(<Player onMinus={onMinusCallback} onPlus={onPlusCallback} />)
    userEvent.click(screen.getByRole('button', { name: '+' }))
    userEvent.click(screen.getByRole('button', { name: '+' }))
    userEvent.click(screen.getByRole('button', { name: '-' }))
    expect(onPlusCallback).toHaveBeenCalledTimes(2)
    expect(onMinusCallback).toHaveBeenCalledTimes(1)
  })

  it('changes the color based on score', () => {
    const { rerender } = render(<Player name="Foo" score={0} />)
    expect(screen.getByText('0')).toMatchSnapshot()
    rerender(<Player score={10} />)
    expect(screen.getByText('10')).toMatchSnapshot()
    rerender(<Player score={-10} />)
    expect(screen.getByText('-10')).toMatchSnapshot()
  })
})
