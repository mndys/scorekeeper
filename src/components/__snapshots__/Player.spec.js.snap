// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player changes the color based on score 1`] = `
.c0 {
  color: black;
}

<span
  class="c0"
  color="black"
>
  0
</span>
`;

exports[`Player changes the color based on score 2`] = `
<span
  class="Player__ColorScore-sc-1uws7bn-2 eecmQb"
  color="hsl(100, 50%, 50%)"
>
  10
</span>
`;

exports[`Player changes the color based on score 3`] = `
<span
  class="Player__ColorScore-sc-1uws7bn-2 dHBoRS"
  color="hsl(0, 50%, 50%)"
>
  -10
</span>
`;
